controller/model/base_manager.go

```diff
@@ -381,6 +383,31 @@ func (self *baseEntityManager) iterateRelatedEntitiesInTx(tx *bbolt.Tx, id, fiel
 	return nil
 }
 
+func (self *baseEntityManager) Annotate(ctx boltz.MutateContext, entityId string, key, value string) error {
+	entityBucket := self.GetStore().GetEntityBucket(ctx.Tx(), []byte(entityId))
+	if entityBucket == nil {
+		return boltz.NewNotFoundError(self.GetStore().GetEntityType(), "id", entityId)
+	}
+	annotationsBucket := entityBucket.GetOrCreatePath(annotationsBucketName)
```

Comment:
So there is a reserved bucket named annotate on every entity that stores a single string?